### PR TITLE
Update cca.py

### PR DESCRIPTION
**Pull request by ricor07**
Rewrote statistic function in cca.py.

Now it outputs similar values to scikitlearn.

For similar i mean values equal up to the third or fourth decimal.

### CODE REVIEW

**Comment by Riccardo231** [comment]
Excuse me, the hyppo test fails even with the code that was already there. How can I be sure my modifies are correct or not?

**Comment by sampan501** [comment]
Can you write a unit test for your change?

**Comment by Riccardo231** [comment]
Done. I don't know if it's what you're looking for

**Comment by sampan501** [comment]
Almost perfect. Instead of changing it as you have with unittest, I think it's sufficient to leave the file as is and only modify the oned test with the one you wrote

**Comment by sampan501** [comment]
If that is successful, I'll merge

**Comment by Riccardo231** [comment]
Good morning. After revising the statistic function, I ran the tests with the original implementation. I observed that 0.07 was not the expected output for a `joint_normal(n, 3)` distribution with seed 123456789.

`np.random.seed(123456789)
cca_sklearn = CCA_sklearn(1)
cca_hyppo = CCA()

x, y = joint_normal(1000, 3)
print(cca_hyppo.test(x, y), cca_hyppo.statistic(x, y))
print(np.corrcoef(*cca_sklearn.fit_transform(x, y), rowvar=False).diagonal(offset=1)[0])`

Both the hyppo and sklearn methods return the same output, which for `n = 1000` is 0.5034560517160103.

I modified the expected statistic values in the tests, and they now pass successfully.

Regarding the errors in the hyppo bot tests, most of them seem to be precision errors. However, as I mentioned earlier, the original CCA code also produced similar errors in these tests, so I am unsure how to proceed.

Please let me know how you would like to address this.

**Comment by Riccardo231** [comment]
Apologies for the disturbance once again. After numerous trials, I was able to identify and resolve the issue with the program. The problem was related to the squaring of values in the initial case. However, during testing, one particular test case failed to produce the correct result. Upon thorough review, I confirmed that the test itself was incorrect and subsequently modified it. Following this correction, all checks now pass successfully. I believe the changes are ready to be merged.
